### PR TITLE
MKL support for the transpose op

### PR DESCRIPTION
**Pull request by vivek-rane**
2D transpose support for float.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_matmul_op.cc [hunk 1]
```diff
@@ -0,0 +1,217 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (vivek-rane) [inline tensorflow/core/kernels/mkl_matmul_op.cc]: This file is part of a different pull request, which has been accepted already and is in the test phase.

File: tensorflow/core/kernels/matmul_op.cc [hunk 2]
```diff
@@ -276,13 +276,20 @@ struct MatMulFunctor<CPUDevice, T> {
                               .Label("cublas"),                    \
                           MatMulOp<GPUDevice, T, true /* cublas */>)
 
+#if defined (INTEL_MKL)
```
Comment (vivek-rane) [inline tensorflow/core/kernels/matmul_op.cc]: This change is part of the matmul pull request and doesn't need to be reviewed here.

**Review by andydavis1 — APPROVED** [review]

**Comment by maciekcc** [comment]
Jenkins, test this please.

**Comment by vivek-rane** [comment]
Why aren't the tests progressing?

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

**Comment by vivek-rane** [comment]
I see a strange timeout error in the Linux CPU Tests console output:

INFO: Timeout connecting to https://github.com/junit-team/junit4/releases/download/r4.12/junit-4.12.jar
Build was aborted
Aborted by unknown
Unable to get pull request builder trigger!!
Setting status of 06c0c6d0c61321de291d0fd96971cc300748579d to FAILURE with url https://ci.tensorflow.org/job/tensorflow-pull-requests-cpu-python3/3537/ and message: 'FAILURE
 '
Using context: Linux CPU Tests (Python 3)
Finished: ABORTED

**Comment by vivek-rane** [comment]
@tensorflow-jenkins test this please

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

**Comment by vivek-rane** [comment]
Does it only listen to you? :)

**Comment by vivek-rane** [comment]
I’m not sure what is going on with the transpose op tests – I looked at the build log but didn’t see any real errors. I was wondering if someone could help look at what is wrong?

There is a buildifier error in the sanity check which I will fix, but for the linux CPU tests for instance, the log (https://ci.tensorflow.org/job/tensorflow-pull-requests-cpu/3899/console) suddenly ends in error with no error message.

Thanks for your help.

**Comment by andydavis1** [comment]
I was just told that the rest of the tests will stop if the sanity check fails, which failed because of this following buildifier error (see below) So if you fix that, things should progress...

=== Sanity check step 3 of 7: do_buildifier (buildifier check) ===

Running do_buildifier on 179 files

tensorflow/core/kernels/BUILD # reformat 

buildifier took 1 s

FAIL: buildifier found errors and/or warnings in above BUILD files.
buildifier suggested the following changes:
694,698c694,697
<         "transpose_op.cc"
<         ]
<         + if_mkl([
<         "mkl_transpose_op.cc"
<         ]),
---
>         "transpose_op.cc",
>     ] + if_mkl([
>         "mkl_transpose_op.cc",
>     ]),
700,703c699,701
<     deps = ARRAY_DEPS
<         + if_mkl([
<                 "//third_party/mkl:intel_binary_blob",
<         ]),
---
>     deps = ARRAY_DEPS + if_mkl([
>         "//third_party/mkl:intel_binary_blob",
>     ]),
Please fix manually or run buildifier <file> to auto-fix.

**Comment by vivek-rane** [comment]
Fixed with buildifier. Looks like the tests have restarted automatically.

**Comment by vivek-rane** [comment]
Looks like the tests need to be restarted. @andydavis1 please could you kick them off?

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

**Comment by vivek-rane** [comment]
The errors here had no relation to my changes, so I rebased to the latest TensorFlow master. Not sure what the deal is with the CLA comment by @googlebot, but please can you restart the tests?

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

**Comment by vivek-rane** [comment]
Restart tests?

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

**Comment by vivek-rane** [comment]
Looks like we're done here :)

**Comment by jhseu** [comment]
@vivek-rane GitHub is showing a large diff from master and this change seems to duplicate a lot of existing commits there. I couldn't figure out why it's doing that. Mind trying to fix it?

Probably the simplest way is to copy the diff out, remake the branch with a fresh checkout from master, and `git push -f` to the branch to update this pull request.

**Comment by vivek-rane** [comment]
@jhseu I don't see the issue you mentioned on the diff page (https://github.com/tensorflow/tensorflow/pull/7748/files) - am I looking at the wrong place?

**Comment by jhseu** [comment]
Looks like your recent master merge fixed it for me too... weird. Looks good now.

Jenkins, test this please.

**Comment by vivek-rane** [comment]
Error in sanity checks - any idea what happened? Testing was interrupted for some reason (I think)

INFO: Loading package: @bazel_tools//tools/zip
java.lang.RuntimeException: Unrecoverable error while evaluating node 'REPOSITORY_DIRECTORY:@org_polymer_paper_spinner' (requested by nodes 'REPOSITORY:@org_polymer_paper_spinner')
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:429)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:501)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: Invalid EvalException:
java.lang.InterruptedException

**Comment by jhseu** [comment]
I'm guessing it's a transient error. Trying again:
Jenkins, test this please

**Comment by vivek-rane** [comment]
Sanity checks still show the same error, but I dug into the ci.tensorflow.org console and it shows this - maybe there's something acting up with Jenkins?

Building image tf-make-base...
Error checking context: 'can't stat '/var/lib/jenkins/workspace/tensorflow-pull-requests-makefile/tensorflow/contrib/makefile/downloads/gemmlowp/test''.
ERROR: docker build failed. Dockerfile is at /var/lib/jenkins/workspace/tensorflow-pull-requests-makefile/tensorflow/contrib/makefile/../../../tensorflow/contrib/makefile/Dockerfile

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by vivek-rane** [comment]
Same errors (interrupted exception)

**Comment by jhseu** [comment]
That's pretty unusual...

Jenkins, test this please

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by jhseu** [comment]
Jenkins, test this please

**Pull request merged**